File: Data Structures & Algorithms/subarrays-with-k-different-integers/submission-1.cpp
class Solution {
private:
    int helper(vector<int>& nums, int k) {
        int l = 0, res = 0;

        unordered_map<int, int> count;
        for (int r = 0; r < nums.size(); ++r) {
            count[nums[r]] += 1;

            while (count.size() > k) {
                count[nums[l]] -= 1;
                if (count[nums[l]] == 0) {
                    count.erase(nums[l]);
                }
                ++l;
            }

            res += r - l + 1;
        }

        return res;
    }
public:
    int subarraysWithKDistinct(vector<int>& nums, int k) {
        return helper(nums, k) - helper(nums, k - 1);
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        count, max_count = 0, 0
        s = set(nums)

        for n in nums:
            if (n - 1) not in s:
                count = 0
                while n in s:
                    count += 1
                    n += 1
                max_count = max(count, max_count)
        return max_count
